src/main/java/org/mskcc/cmo/messaging/impl/JSGatewayImpl.java

```diff
@@ -297,4 +296,43 @@ public PublishingQueueTask(String msgId, String subject, Object payload) {
             this.payload = payload;
         }
     }
+
+    @Override
+    public String request(String subject, Object message) {
+        if (natsConnection == null) {
```

Comment:
in the subscribe and publish methods we are checking the natsConnection with another method `isConnected()` - we should keep this consistent

Reply:
Will add that!